src/app.test.js

```diff
@@ -1,15 +1,52 @@
-// import { createApp } from "./app";
-// import userevent from '@testing-library/user-event'
-
-test("renders the todo app", () => {
-  //   // setup
-  //   const root = ...
-  //   const container = createApp(root)
-  //   // act
-  //   const input = queryByPlaceholderText('Please enter todo here')
-  //   userevent.type(input, 'new task')
-  //   const addButton = queryByText('Add')
-  //   userevent('click', addButton)
-  //   // assert
-  expect(true).tobee(true);
+import { screen, getByText } from "@testing-library/dom";
+import userEvent from "@testing-library/user-event";
+import createApp from "./app";
+// import userEvent from "@testing-library/user-event";
+
+test("renders empty todo list when starts", () => {
+  // setup
+  const root = document.body;
+  createApp(root);
+  const todoList = screen.getByTestId("todo-list");
+  // act
+  // assert
+  expect(todoList).toBeEmptyDOMElement();
+});
+
+test("shows new item in todo list when added", () => {
+  // setup
+  const input = screen.getByPlaceholderText("Please enter todo here");
+  const addButton = screen.getByTestId("add-button");
+  const inputText = "Take wife for a walk!";
+
+  // act
+  userEvent.type(input, inputText);
+  userEvent.click(addButton);
+
+  // assert
+  const todoList = screen.getByTestId("todo-list");
+  const todoTextElement = getByText(todoList, "Take wife for a walk!");
+  expect(todoTextElement).toHaveTextContent(inputText);
```

Comment:
we don't need to find the "text element", instead we should find the new todo `"Take wife for a walk!"` by its text:

```js
expect(queryByText(todoList, "Take wife for a walk!")).not.toBeNull();
```

this is better as it takes us closer to how the user would find it on the screen

Reply:
eslint does not like to use ToBeNull : 
This is the same as .toBe(null) but the error messages are a bit nicer. So use .toBeNull() when you want to check that something is null.
Prefer .toBeInTheDocument() for asserting DOM node existence